### PR TITLE
YTA-3215: Added endpoint for Payment History with Allocations.

### PR DESCRIPTION
**Pull request by cybersaurus**

### CODE REVIEW

File: app/uk/gov/hmrc/epayeapi/models/in/EpayeTaxPeriod.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    implicit val dateAscendingOrdering = uk.gov.hmrc.epayeapi.models.ImplicitOrderings.localDateDescendingOrdering.reverse
+
+    val Seq(lowerDate, upperDate) = Seq(taxFrom, taxTo).sorted
+    val expectedTo1 = lowerDate.plusMonths(1).minusDays(1)
+    val expectedTo2 = lowerDate.plusMonths(1)
+    val isTaxToCorrect1 = expectedTo1 == upperDate
+    val isTaxToCorrect2 = expectedTo2 == upperDate
+    (lowerDate.getMonthOfYear, lowerDate.getDayOfMonth) match {
+      case (m, 6) if isTaxToCorrect1 => Some(((m + 8) % 12) + 1)
+      case (m, 4) if isTaxToCorrect2 => Some(((m + 8) % 12) + 1)
```
Comment (flashingpumpkin) [inline app/uk/gov/hmrc/epayeapi/models/in/EpayeTaxPeriod.scala]: Can we remove this exception? This was due to invalid data in QA and should never happen.

Comment (cybersaurus) [reply app/uk/gov/hmrc/epayeapi/models/in/EpayeTaxPeriod.scala]: Will do. This was copied from the backend. Should we remove the exception from there as well?

File: resources/public/api/conf/1.0/schemas/PaymentHistoryWithAllocations.schema.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          "items": {
+            "type": "object",
+            "properties": {
+              "taxYear": {
+                "$ref": "Definitions.schema.json#/definitions/taxYear"
+              },
+              "taxMonth": {
+                "$ref": "Definitions.schema.json#/definitions/taxMonth"
+              }
+            },
```
Comment (jose-puente-digital-hmrc-gov-uk) [inline resources/public/api/conf/1.0/schemas/PaymentHistoryWithAllocations.schema.json]: Should have an amount field

File: resources/public/api/conf/1.0/schemas/PaymentHistoryWithAllocations.schema.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      "properties": {
+        "paymentDate": {
+          "description": "The date the payment was received",
+          "$ref": "Definitions.schema.json#/definitions/date"
+        },
+        "amount": {
+          "description": "The first day of the tax year",
+          "type": "number"
+        },
+        "method": {
```
Comment (jose-puente-digital-hmrc-gov-uk) [inline resources/public/api/conf/1.0/schemas/PaymentHistoryWithAllocations.schema.json]: Should have an unallocatedAmount field

File: resources/public/api/conf/1.0/schemas/PaymentHistoryWithAllocations.schema.json [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      "title": "Description of a payment and allocations received",
+      "properties": {
+        "paymentDate": {
+          "description": "The date the payment was received",
+          "$ref": "Definitions.schema.json#/definitions/date"
+        },
+        "amount": {
+          "description": "The first day of the tax year",
+          "type": "number"
+        },
```
Comment (jose-puente-digital-hmrc-gov-uk) [inline resources/public/api/conf/1.0/schemas/PaymentHistoryWithAllocations.schema.json]: should have an unallocatedAmount field

**Pull request merged**